packages/@aws-cdk/aws-ecs/README.md

```diff
@@ -1,2 +1,142 @@
-## The CDK Construct Library for AWS Elastic Container Service (ECS)
-This module is part of the [AWS Cloud Development Kit](https://github.com/awslabs/aws-cdk) project.
+## AWS Elastic Container Service (ECS) Construct Library
+
+This package contains constructs for working with **AWS Elastic Container
+Service** (ECS). The simplest example of using this library looks like this:
+
+```ts
+// Create an ECS cluster (backed by an AutoScaling group)
+const cluster = new ecs.EcsCluster(this, 'Cluster', {
```

Comment:
Do we need two separate constructs for `ecs.EcsCluster` and `ecs.FargateCluster`? I think this introduces some naming complexity (especially later down the line when we have Fargate for EKS). By having them so separate, it also hides the fact that customers can (and do) have mixed-mode clusters with some EC2 and some Fargate mode tasks.

The only real difference is how many instances (if any) are configured. Why not have a single `new ecs.Cluster()` construct, that accepts an optional `instanceConfiguration` prop.

Reply:
While it's true that we wanted to supported mixed-mode clusters, these are an extremely low percentage of customers. We wanted the construct library to provide a more opinionated way to set up infrastructure. In addition provisioning capacity in an ECS cluster, there is some differences in the networking setup as well.

Unless you know of good reasons why customers wouldn't want to have a homogeneous cluster, keeping separate cluster types seemed to streamline the design of other ECS constructs.